src/libraries/System.Net.Sockets/src/System/Net/Sockets/Socket.cs

```diff
@@ -1269,10 +1269,36 @@ public int Send(ReadOnlySpan<byte> buffer, SocketFlags socketFlags, out SocketEr
 
         public void SendFile(string? fileName)
         {
-            SendFile(fileName, null, null, TransmitFileOptions.UseDefaultWorkerThread);
+            SendFile(fileName, ReadOnlySpan<byte>.Empty, ReadOnlySpan<byte>.Empty, TransmitFileOptions.UseDefaultWorkerThread);
         }
 
         public void SendFile(string? fileName, byte[]? preBuffer, byte[]? postBuffer, TransmitFileOptions flags)
+        {
+            SendFile(fileName, preBuffer.AsSpan(), postBuffer.AsSpan(), flags);
+        }
+
+        /// <summary>
+        /// Sends the file <paramref name="fileName"/> and buffers of data to a connected <see cref="Socket"/> object
+        /// using the specified <see cref="TransmitFileOptions"/> value.
+        /// </summary>
+        /// <param name="fileName">
+        /// A <see cref="string"/> that contains the path and name of the file to be sent. This parameter can be <see langword="null"/>.
+        /// </param>
+        /// <param name="preBuffer">
+        /// A <see cref="byte"/> array that contains data to be sent before the file is sent. This parameter can be <see langword="null"/>.
```

Comment:
This comment is wrong in multiple ways -- it's not a byte array, it can't be null.

Is this just copied from existing docs? If so, please add it to the existing overloads, and then revise this one to be appropriate for the new overload.

Reply:
> Is this just copied from existing docs?

Ui, yes and now for this PR from the old. I feel ashamed that this happened. Thanks.

Reply:
We're in kind of a weird space currently for changes like this, i.e. adding a new overload to a method that has docs on MSDN but not in the source. Best policy is to copy the existing docs for the existing overloads into the source manually, then add/adapt that for the new overload(s) too.